File: src/main/java/no/nav/bidrag/beregn/felles/FellesPeriode.java
package no.nav.bidrag.beregn.felles;

import java.time.LocalDate;
import java.util.List;
import no.nav.bidrag.beregn.felles.bo.Periode;

public abstract class FellesPeriode {

  protected void mergeSluttperiode(List<Periode> periodeListe, LocalDate datoTil) {
    if (periodeListe.size() > 1) {
      if ((periodeListe.get(periodeListe.size() - 2).getDatoTil().equals(datoTil)) &&
          (periodeListe.get(periodeListe.size() - 1).getDatoTil() == null)) {
        var nyPeriode = new Periode(periodeListe.get(periodeListe.size() - 2).getDatoFom(), null);
        periodeListe.remove(periodeListe.size() - 1);
        periodeListe.remove(periodeListe.size() - 1);
        periodeListe.add(nyPeriode);
      }
    }
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/felles/FellesPeriode.kt
package no.nav.bidrag.beregn.felles

import no.nav.bidrag.beregn.felles.bo.Periode
import java.time.LocalDate

abstract class FellesPeriode {
    protected fun mergeSluttperiode(periodeListe: MutableList<Periode>, datoTil: LocalDate) {
        if (periodeListe.size > 1) {
            val nestSisteTilDato = periodeListe[periodeListe.size - 2].datoTil
            val sisteTilDato = periodeListe[periodeListe.size - 1].datoTil
            if (datoTil == nestSisteTilDato && null == sisteTilDato) {
                val nyPeriode = Periode(datoFom = periodeListe[periodeListe.size - 2].datoFom, datoTil = null)
                periodeListe.removeAt(periodeListe.size - 1)
                periodeListe.removeAt(periodeListe.size - 1)
                periodeListe.add(nyPeriode)
            }
        }
    }
}
